src/sentry/grouping/enhancer/__init__.py

```diff
@@ -144,10 +164,50 @@ def apply_modifications_to_frame(
         extra_fingerprint: str = "",
     ) -> None:
         """This applies the frame modifications to the frames itself. This does not affect grouping."""
-        in_memory_cache: dict[str, str] = {}
 
         # Matching frames are used for matching rules
         match_frames = [create_match_frame(frame, platform) for frame in frames]
+
+        rust_enhanced_frames = None
+        if enhancements := self.rust_enhancements:
+            e = exception_data or {}
+            e = {
+                "ty": e.get("type"),
+                "value": e.get("value"),
+                "mechanism": get_path(e, "mechanism", "type"),
+            }
+            for key in e.keys():
+                value = e[key]
+                if isinstance(value, str):
+                    e[key] = value.encode("utf-8")
+
+            try:
+                rust_enhanced_frames = enhancements.apply_modifications_to_frames(
+                    iter(match_frames), e
+                )
+            except Exception:
+                logger.exception("failed running Rust Enhancements modifications")
+
+            # TODO: once ready, update the frame directly
```

Comment:
That is the requirement of being ready?

Reply:
I would say when we've determined that the rust version gives the same result and performs adequately (the latter of which is almost a given).